pike/ntstatus.py

```diff
@@ -1,6 +1,13 @@
 from __future__ import absolute_import
+import contextlib
 from . import core
 
+@contextlib.contextmanager
+def pike_status(exp_status):
+    with pytest.raises(pike.model.ResponseError) as excinfo:
```

Comment:
pike doesn't directly depend on `pytest`, additionally `pytest` is not imported in this module.

I think we need an implementation for this that will work for both pytest and unittest. 🤔

maybe it could be as simple as

```
try:
    yield
    if exp_status != STATUS_SUCCESS:
        raise AssertionError("Did not raise ResponseError({})".format(exp_status))
except pike.model.ResponseError as err:
    assert err.response.status == exp_status
```

This has a nice side benefit that the caller can pass `exp_status=STATUS_SUCCESS` to effectively short-circuit the check (this wasn't possible with `PikeTest.assert_error`). Because conditionals and contextmanagers don't mix very well, encapsulating this common parametrization is probably a win.

Reply:
So, looking a bit closer at the existing implementation: https://github.com/emc-isilon/pike/blob/master/pike/test/__init__.py#L435-L449

it probably makes sense to yield some object to the caller so that when this contextmanager catches the exception, it can set an attribute on that object and effectively pass the exception back for further inspection.

```
class AssertNtstatusContext(object):
    # real implementation would define public fields here
    pass

def assert_ntstatus(exp_status):
    o = AssertNtstatusContext()

    try:
        yield o
        if exp_status != STATUS_SUCCESS:
            raise AssertionError("Did not raise ResponseError({})".format(exp_status))
    except pike.model.ResponseError as err:
        o.exception = err
        o.response = err.response
        # maybe use future.utils.raise_from to raise the AssertionError along with the original exc
        assert err.response.status == exp_status
```